server/routes/noteRouter.ts

```diff
@@ -0,0 +1,251 @@
+          body: {
+            paragraphs: notebookinfo,
+          },
+        });
+      }
+    }
+  );
+
+  // Add a Notebook
+  router.post(
+    {
+      path: `${API_PREFIX}/note`,
+      validate: {
+        body: schema.object({
+          name: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response): Promise<IKibanaResponse<any | ResponseError>> => {
+      let addResponse = {};
+      try {
+        addResponse = await BACKEND.addNote(context, request.body, wreckOptions);
+        return response.ok({
+          body: addResponse,
+        });
+      } catch (error) {
+        return response.custom({
+          statusCode: error.statusCode || 500,
+          body: {
+            body: addResponse,
+          },
+        });
+      }
+    }
+  );
+
+  // Rename a notebook
+  router.put(
+    {
+      path: `${API_PREFIX}/note/rename`,
+      validate: {
+        body: schema.object({
+          name: schema.string(),
+          noteId: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response): Promise<IKibanaResponse<any | ResponseError>> => {
+      let renameResponse = {};
+      try {
+        renameResponse = await BACKEND.renameNote(context, request.body, wreckOptions);
+        return response.ok({
+          body: renameResponse,
+        });
+      } catch (error) {
+        return response.custom({
+          statusCode: error.statusCode || 500,
+          body: {
+            body: renameResponse,
+          },
+        });
+      }
+    }
+  );
+
+  // Clone a notebook
+  router.post(
+    {
+      path: `${API_PREFIX}/note/clone`,
+      validate: {
+        body: schema.object({
+          name: schema.string(),
+          noteId: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response): Promise<IKibanaResponse<any | ResponseError>> => {
+      let cloneResponse = {};
+      try {
+        const cloneResponse = await BACKEND.cloneNote(context, request.body, wreckOptions);
+        return response.ok({
+          body: cloneResponse,
+        });
+      } catch (error) {
+        return response.custom({
+          statusCode: error.statusCode || 500,
+          body: {
+            body: cloneResponse,
+          },
+        });
+      }
+    }
+  );
+
+  // Delete a notebook
+  router.delete(
+    {
+      path: `${API_PREFIX}/note/{noteid}`,
+      validate: {
+        params: schema.object({
+          noteid: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response): Promise<IKibanaResponse<any | ResponseError>> => {
+      let delResponse = {};
+      try {
+        delResponse = await BACKEND.deleteNote(context, request.params.noteid, wreckOptions);
+        return response.ok({
+          body: delResponse,
+        });
+      } catch (error) {
+        return response.custom({
+          statusCode: error.statusCode || 500,
+          body: {
+            body: delResponse,
+          },
```

Comment:
Can we add custom error message back to the user side, like to the console or to the UI? Same to other routes

Reply:
Modified implementation for propagate error to UI. Thanks